p2p-specs/p2p-interface.md

```diff
@@ -390,7 +390,7 @@ All messages that contain only a single field MUST be encoded directly as the ty
 Request, Response Content:
 ```
 (
-  List[bytes32,MAX_OPS_PER_REQUEST]
+  List[bytes46,MAX_OPS_PER_REQUEST]
```

Comment:
1. there is no such type as bytes46. 
2. no description in the text for the meaning of this value
3. probably you put together 2 separate values, so document them as such, instead of opaque constant.
4. (and btw, even the current definition is missing explanation as to what this bytes32 value is..)

Reply:
Lookin at this: https://github.com/multiformats/cid/blob/ef1b2002394b15b1e6c26c30545fd485f2c4c138/README.md#decoding-algorithm

We should use `bytes` here, since the length can be different across version numbers. We should define a maximum value for the bytes, I'd suggest `256` for future proofing. We should also refer to this is the IPFS CID instead of IPFS Hash.

https://docs.ipfs.tech/concepts/content-addressing/#what-is-a-cid

https://github.com/multiformats/cid/blob/ef1b2002394b15b1e6c26c30545fd485f2c4c138/README.md#decoding-algorithm

CID decoding is supported in multiple languages if bundlers want to decode (although I don't think they'll need to).

This `bytes` is the UTF-8 encoding. In the `topic` below we use the string version.

Reply:
yes, we agree it is a dynamic value, with a maximum length (of say 256)
why should we call it "bytes" and not "string" ? (not that I really care. either is good, as long as the text explain it is ipfs cid..

Reply:
I think we should keep it as bytes, we use a similar encoding for `error_message`:

```
The ErrorMessage schema is:

  (
    error_message: List[byte, 256]
  )

Note: By convention, the error_message is a sequence of bytes that MAY be interpreted as a UTF-8 string (for debugging purposes). Clients MUST treat as valid any byte sequences.
```

So for this encoding I'd say we use:

```
supported_mempools: List<List<byte, MAX_IPFS_CID_LENGTH>, MAX_SUPPORTED_MEMPOOLS>
```

Where each entry in the mempool list is a sequence of bytes that MUST be interpreted as a UTF-8 string representing an IPFS CID.